packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -562,6 +575,51 @@ function renderUlimit(ulimit: Ulimit): CfnTaskDefinition.UlimitProperty {
     hardLimit: ulimit.hardLimit,
   };
 }
+/**
+ * Container dependency
+ */
+export interface ContainerDependency {
+  /**
+   * The container to depend on
+   */
+  readonly container: ContainerDefinition,
+  /**
+   * The dependency condition of the container
+   */
+  readonly condition?: ContainerDependencyCondition,
```

Comment:
As per https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerDependency.html condition is required, so you should either make condition required or provide a good default (preferable)